### DISCUSSION TITLE
How to actually stop a debugging session?

### DISCUSSION THREAD

**Discussion by tomaskallup** [mfussenegger/nvim-dap#328] [Q&A]
Right now I have following mapping:
```lua
vim.api.nvim_set_keymap("n", "[DAP]e",
                        "<cmd>lua require'dap'.disconnect()<CR><cmd>lua require'dap'.close()<CR>",
                        {})
```
This however only disconnects from the session and keeps the app running.

I'm using this with typescript (but I have custom defined adapter, which builds the app into JS before launching it) and the debug adapter (vscode-node-debug2) doesn't seem to support `restart`, so I have to close the session, manually switch to the terminal with the app and `<C-c>` two times to exit it, so that I can start it again.

**Comment by mfussenegger** [comment]
Did you try with `lua require'dap'.disconnect({ terminateDebuggee = true })` ?

**Reply by tomaskallup** [reply]
Nothing seems to happen.
My new mapping:
```lua
vim.api.nvim_set_keymap("n", "[DAP]e",
                        "<cmd>lua require'dap'.disconnect({ terminateDebuggee = true })<CR><cmd>lua require'dap'.close()<CR>",
                        {})
```

**Reply by mfussenegger** [reply]
Sounds like a vscode-node-debug2 problem then.
